src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -44,23 +45,28 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
       .endpoint(getEndpoint)
       .buildClient()
 
-  import spark.implicits._
-  lazy val df: DataFrame = Seq(
-    (Seq(""),Seq("")),
-  ).toDF("lang", "text")
+  val inputSchema = StructType(Array(
+    StructField("lang",ArrayType(StringType,true),true),
+    StructField("text",ArrayType(StringType,true),true)
+  ))
+
+  setDefault(batchSize -> 5)
 
   protected def transformTextRows(toRow: TAResponseV4[T] => Row)
                                  (rows: Iterator[Row]): Iterator[Row] = {
-
-    rows.map { row =>
-      val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
-      Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
-    }}
+    val futures = rows.map { row =>
+      Future {
+        val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
+        Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
+      }(ExecutionContext.global)
+    }
+    bufferedAwait(futures, getConcurrency, Duration(getTimeout,SECONDS))(ExecutionContext.global)
+  }
 
   override def transform(dataset: Dataset[_]): DataFrame = {
-    val batchedDF = new FixedMiniBatchTransformer().setBatchSize(10).transform(dataset.coalesce(1))
-    val finaldataset = spark.createDataFrame(batchedDF.rdd, df.schema)
     logTransform[DataFrame]({
+      val batchedDF = new FixedMiniBatchTransformer().setBatchSize(getBatchSize).transform(dataset.coalesce(1))

```

Comment:
Do we really need the `coalesce(1)`?

Reply:
fails without it --> this is how mark called it in TASuite